File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
var findMaxAverage = function (nums, k) {
    if (nums.length === 1) return nums[0];
    let windowSum = 0;
    for (let i = 0; i < k; i++) {
        windowSum += nums[i];
    }
    let MaxSum = windowSum;
    for (let i = k; i < nums.length; i++) {
        windowSum = windowSum - nums[i - k] + nums[i]
        if (windowSum > MaxSum) MaxSum = windowSum;
    }
    return MaxSum / k
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cs
public class Solution {
    public double FindMaxAverage(int[] nums, int k) {
        double MaxSum = 0;

        for(int i = 0; i < k; i++) MaxSum += nums[i];

        double temp = MaxSum;

        for(int j = k; j < nums.Length; j++)
        {
            temp += nums[j] - nums[j - k];
            if(temp > MaxSum) MaxSum = temp;
        }

        return MaxSum / k;
    }
}